phantazm-zombies/src/main/java/com/github/phantazmnetwork/zombies/equipment/gun/shoot/handler/GuardianBeamShotHandler.java

```diff
@@ -0,0 +1,131 @@
+public class GuardianBeamShotHandler implements ShotHandler {
+
+    public record Data(@NotNull EntityType entityType, long beamTime) implements Keyed {
+
+        public static final Key SERIAL_KEY = Key.key(Namespaces.PHANTAZM, "gun.shot_handler.guardian_beam");
+
+        public Data {
+            Objects.requireNonNull(entityType, "entityType");
+            if (beamTime < 0) {
+                throw new IllegalArgumentException("beamTime must be greater than or equal to 0");
+            }
+        }
+
+        @Override
+        public @NotNull Key key() {
+            return SERIAL_KEY;
+        }
+    }
+
+    public static @NotNull ConfigProcessor<Data> processor() {
+        return new ConfigProcessor<>() {
+
+            @Override
+            public @NotNull Data dataFromElement(@NotNull ConfigElement element) throws ConfigProcessException {
+                boolean isElder = element.getBooleanOrThrow("isElder");
+                long beamTime = element.getNumberOrThrow("beamTime").longValue();
+                if (beamTime < 0) {
+                    throw new ConfigProcessException("beamTime must be greater than or equal to 0");
+                }
+
+                return new Data(isElder ? EntityType.ELDER_GUARDIAN : EntityType.GUARDIAN, beamTime);
+            }
+
+            @Override
+            public @NotNull ConfigElement elementFromData(@NotNull Data data) {
+                ConfigNode node = new LinkedConfigNode(1);
+                node.putBoolean("isElder", data.entityType() == EntityType.ELDER_GUARDIAN);
+                node.putNumber("beamTime", data.beamTime());
+                return node;
+            }
+        };
+    }
+
+    private record Beam(@NotNull Reference<Instance> instance, @NotNull Entity guardian, @NotNull Entity armorStand,
+                        long time) {
+
+    }
+
+    private final Queue<Beam> removalQueue = new PriorityQueue<>(Comparator.comparingLong(Beam::time));
+
+    private final Data data;
+
+    public GuardianBeamShotHandler(@NotNull Data data) {
+        this.data = Objects.requireNonNull(data, "data");
+    }
+
+    @Override
+    public void handle(@NotNull GunState state, @NotNull Entity attacker, @NotNull Collection<UUID> previousHits, @NotNull GunShot shot) {
+        Instance instance = attacker.getInstance();
+        if (instance == null) {
+            return;
+        }
+
+        Entity armorStand = new Entity(EntityType.ARMOR_STAND);
+        EntityMeta armorStandMeta = armorStand.getEntityMeta();
+        armorStandMeta.setInvisible(true);
+
+        Entity guardian = new Entity(data.entityType());
+        GuardianMeta guardianMeta = (GuardianMeta) guardian.getEntityMeta();
+        guardianMeta.setTarget(armorStand);
+        guardian.setInvisible(true);
+
+        Pos start = attacker.getPosition().add(0, attacker.getEyeHeight(), 0);
+        ServerPacket armorStandSpawnPacket = new SpawnLivingEntityPacket(armorStand.getEntityId(), armorStand.getUuid(),
```

Comment:
i understand needing guardian packets, but why armorstands?

Reply:
(also a guardian beam API would be nice in case we want to use it for other things, like mobs)

Reply:
when guardians shoot they need an entity to target